File: app/src/main/java/com/example/qintong/animatordemo/DemoActivity.java
package com.example.qintong.animatordemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;
import com.qintong.library.InsLoadingView;

import static android.widget.Toast.LENGTH_SHORT;

public class DemoActivity extends AppCompatActivity {
    InsLoadingView mInsLoadingView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mInsLoadingView = (InsLoadingView) findViewById(R.id.loading_view);
/*        mInsLoadingView.setCircleDuration(2000);
        mInsLoadingView.setRotateDuration(10000);
        mInsLoadingView.setStartColor(Color.YELLOW);
        mInsLoadingView.setEndColor(Color.BLUE);*/
        mInsLoadingView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (mInsLoadingView.getStatus()) {
                    case UNCLICKED:
                        mInsLoadingView.setStatus(InsLoadingView.Status.LOADING);
                        break;
                    case LOADING:
                        mInsLoadingView.setStatus(InsLoadingView.Status.CLICKED);
                        break;
                    case CLICKED:
                        mInsLoadingView.setStatus(InsLoadingView.Status.UNCLICKED);
                }
                Toast.makeText(DemoActivity.this,"click !", LENGTH_SHORT).show();
            }
        });
        mInsLoadingView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                Toast.makeText(DemoActivity.this,"long click !", LENGTH_SHORT).show();
                return true;
            }
        });
    }
}

File: app/src/main/java/com/example/qintong/animatordemo/DemoActivity.kt
package com.example.qintong.animatordemo

import android.graphics.Color
import android.os.Bundle
import android.widget.Toast
import android.widget.Toast.LENGTH_SHORT
import androidx.appcompat.app.AppCompatActivity
import com.qintong.library.InsLoadingView
import kotlinx.android.synthetic.main.activity_main.*

class DemoActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loading_view?.setCircleDuration(2000)
        loading_view?.setRotateDuration(10000)
        loading_view?.setStartColor(Color.YELLOW)
        loading_view?.setEndColor(Color.BLUE)
        loading_view?.setOnClickListener {
            when (loading_view?.status) {
                InsLoadingView.Status.UNCLICKED -> loading_view?.status = InsLoadingView.Status.LOADING
                InsLoadingView.Status.LOADING -> loading_view?.status = InsLoadingView.Status.CLICKED
                InsLoadingView.Status.CLICKED -> loading_view?.status = InsLoadingView.Status.UNCLICKED
            }
            Toast.makeText(this@DemoActivity, "click !", LENGTH_SHORT).show()
        }
        loading_view?.setOnLongClickListener {
            Toast.makeText(this@DemoActivity, "long click !", LENGTH_SHORT).show()
            true
        }
    }
}
